src/libstd/f32.rs

```diff
@@ -192,6 +192,38 @@ impl f32 {
         }
     }
 
+    /// Returns the maximum of the two numbers.
+    ///
+    /// ```
+    /// let x = 1.0f32;
+    /// let y = 2.0f32;
+    ///
+    /// assert_eq!(x.max(y), y);
+    /// ```
+    ///
+    /// If one of the arguments is NaN, then the other argument is returned.
+    #[stable(feature = "rust1", since = "1.0.0")]
+    #[inline]
+    pub fn max(self, other: f32) -> f32 {
+        unsafe { intrinsics::maxnumf32(self, other) }
+    }
+
+    /// Returns the minimum of the two numbers.
+    ///
+    /// ```
+    /// let x = 1.0f32;
+    /// let y = 2.0f32;
+    ///
+    /// assert_eq!(x.min(y), x);
+    /// ```
+    ///
+    /// If one of the arguments is NaN, then the other argument is returned.
+    #[stable(feature = "rust1", since = "1.0.0")]
+    #[inline]
+    pub fn min(self, other: f32) -> f32 {
+        unsafe { intrinsics::minnumf32(self, other) }
```

Comment:
Do these need to be unsafe?

Reply:
Intrinsics have to be unsafe.

Reply:
Nope. See https://github.com/rust-lang/rust/blob/master/src/librustc_typeck/check/intrinsic.rs#L65 and https://github.com/rust-lang/rust/pull/57234/files.

Reply:
Ah, I didn't realise this changed. These probably shouldn't need to be unsafe, then.